lib/active_merchant/billing/gateways/card_connect.rb

```diff
@@ -0,0 +1,283 @@
+        post[:retref] = authorization
+      end
+
+      def add_additional_data(post, options)
+        post[:ponumber] = options[:po_number]
+        post[:taxamnt] = options[:tax_amount] if options[:tax_amount]
+        post[:frtamnt] = options[:freight_amount] if options[:freight_amount]
+        post[:dutyamnt] = options[:duty_amount] if options[:duty_amount]
+        post[:orderdate] = options[:order_date] if options[:order_date]
+        post[:shipfromzip] = options[:ship_from_zip] if options[:ship_from_zip]
+        if (shipping_address = options[:shipping_address])
+          post[:shiptozip] = shipping_address[:zip]
+          post[:shiptocountry] = shipping_address[:country]
+        end
+        if options[:items]
+          post[:items] = []
+          options[:items].each do |item|
+            updated = {}
+            item.each_pair do |k, v|
+              updated.merge!(k.to_s.gsub(/_/, '') => v)
+            end
+            post[:items] << updated
+          end
+        end
+      end
+
+      def add_3DS(post, options)
+        post[:secureflag] = options[:secure_flag] if options[:secure_flag]
+        post[:securevalue] = options[:secure_value] if options[:secure_value]
+        post[:securexid] = options[:secure_xid] if options[:secure_xid]
+      end
+
+      def headers
+        {
+          'Authorization' => 'Basic ' + Base64.strict_encode64("#{@options[:username]}:#{@options[:password]}").strip,
+          'Content-Type' => 'application/json'
+        }
+      end
+
+      def expdate(credit_card)
+        "#{format(credit_card.month, :two_digits)}#{format(credit_card.year, :two_digits)}"
+      end
+
+      def parse(body)
+        JSON.parse(body)
+      end
+
+      def url(action)
+        if test?
+          test_url + action
+        else
+          (@options[:domain] ? @options[:domain] : live_url) + action
+        end
+      end
+
+      def commit(action, parameters)
+        parameters[:merchid] = @options[:merchant_id]
+        url = url(action)
+        response = parse(ssl_request(:put, url, post_data(parameters), headers))
+
+        Response.new(
+          success_from(response),
+          message_from(response),
+          response,
+          authorization: authorization_from(response),
+          avs_result: AVSResult.new(code: response['avsresp']),
+          cvv_result: CVVResult.new(response['cvvresp']),
+          test: test?,
+          error_code: error_code_from(response)
+        )
+      end
+
+      def success_from(response)
+        response['respstat'] == 'A'
+      end
+
+      def message_from(response)
+        response['setlstat'] ? "#{response['resptext']} #{response['setlstat']}" : response['resptext']
+      end
+
+      def authorization_from(response)
+        response['retref']
+      end
+
+      def post_data(parameters = {})
+        parameters.to_json
+      end
+
+      def error_code_from(response)
+        return false if success_from(response)
```

Comment:
Most of our other gateways return `nil` here, which might allow you to just drop this line in favor of a slight tweak to the next.